src/components/CytoscapeGraph/graphs/GraphStyles.ts

```diff
@@ -200,17 +205,47 @@ export class GraphStyles {
     const isOutside = node.isOutside;
 
     let icons = '';
+    // Put a little more space between icons when a badge has multiple icons
+    const iconMargin = () => (icons === '' ? style(defaultIconMargin) : style({ marginRight: '2px' }));
     if (node.isRoot) {
-      icons = `<span class="${NodeIconRoot} ${iconMargin}"></span> ${icons}`;
+      icons = `<span class="${NodeIconRoot} ${iconMargin()}"></span> ${icons}`;
     }
     if (cyGlobal.showMissingSidecars && node.hasMissingSC) {
-      icons = `<span class="${NodeIconMS} ${iconMargin}"></span> ${icons}`;
-    }
-    if (cyGlobal.showCircuitBreakers && node.hasCB) {
-      icons = `<span class="${NodeIconCB} ${iconMargin}"></span> ${icons}`;
+      icons = `<span class="${NodeIconMS} ${iconMargin()}"></span> ${icons}`;
     }
     if (cyGlobal.showVirtualServices && node.hasVS) {
-      icons = `<span class="${NodeIconVS} ${iconMargin}"></span> ${icons}`;
+      // If there's an additional traffic scenario present then it's assumed
+      // that there is a VS present so the VS badge is omitted.
+      const hasKialiScenario =
+        node.hasCB ||
+        node.hasFaultInjection ||
+        node.hasRequestRouting ||
+        node.hasRequestTimeout ||
+        node.hasTCPTrafficShifting ||
+        node.hasTrafficShifting;
+      if (!hasKialiScenario) {
+        icons = `<span class="${NodeIconVS} ${iconMargin()}"></span> ${icons}`;
+      } else {
+        if (node.hasRequestRouting) {
+          icons = `<span class="${NodeIconRequestRouting} ${iconMargin()}"></span> ${icons}`;
+        }
+        if (node.hasFaultInjection) {
+          icons = `<span class="${NodeIconFaultInjection} ${iconMargin()}"></span> ${icons}`;
+        }
+        if (node.hasTrafficShifting) {
+          icons = `<span class="${NodeIconTrafficShifting} ${iconMargin()}"></span> ${icons}`;
+        }
+        // Don't show icon twice since the same for both http and tcp traffic shifting
+        if (node.hasTCPTrafficShifting && !node.hasTrafficShifting) {
+          icons = `<span class="${NodeIconTCPTrafficShifting} ${iconMargin()}"></span> ${icons}`;
+        }
```

Comment:
If you take the recommendation to remove `NodeIconTCPTrafficShifting` then this can just be:

```
if (node.hasTrafficShifting || node.hasTCPTrafficShifting) {
...
```